src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Design/DefineAccessorsForAttributeArguments.cs

```diff
@@ -22,22 +22,44 @@ public abstract class DefineAccessorsForAttributeArgumentsAnalyzer : DiagnosticA
         internal const string AddAccessorCase = "AddAccessor";
         internal const string MakePublicCase = "MakePublic";
         internal const string RemoveSetterCase = "RemoveSetter";
-        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArguments), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
 
-        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
-                                                                         s_localizableTitle,
-                                                                         "{0}",
-                                                                         DiagnosticCategory.Design,
-                                                                         DiagnosticSeverity.Warning,
-                                                                         isEnabledByDefault: false,
-                                                                         helpLinkUri: "http://msdn.microsoft.com/library/ms182136.aspx",
-                                                                         customTags: WellKnownDiagnosticTags.Telemetry);
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArguments), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+        private static readonly LocalizableString s_defaultRuleMessage = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArgumentsDefault), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+        private static readonly LocalizableString s_increaseVisiblityMessage = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArgumentsIncreaseVisibility), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+        private static readonly LocalizableString s_removeSetterMessage = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArgumentsRemoveSetter), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+
+        internal static DiagnosticDescriptor DefaultRule = new DiagnosticDescriptor(RuleId,
```

Comment:
@heejaechang Note that this will break compat with some FXCop rules, which use same ID for different messages/diagnostics.

Unless we create different analyzers for each diagnostic, but that doesn't help either as now we have multiple analyzers with descriptors having same Id.

Btw, @heejaechang I remember you had opened a TFS issue about the same thing: "each analyzer must return descriptor with unique ID", and I think triage had Won't fixed it. Did your proposal later get approved?